### DISCUSSION TITLE
Grouping categories together to have an aggregate view of spending ?

### DISCUSSION THREAD

**Discussion by R-Nabil** [firefly-iii/firefly-iii#9124] [General]
Hi,

I'd like to organise my transaction into categories and the group of categories. 

For instance, Restaurant&Drinks would be the aggregate of Restaurants, Drinks, Coffees, Snacks subcategories. 

I thought this would be achievable by doing a category for each and then assigning them to a budget but budget seems to require an amount (as it is a budget) and I do not need to set one. 

Idea is to be able to have an overview of that "aggregate" and be able to see on a monthly basis what is the reason for an overspending or underspending on it

Is there a way to achieve that ?

**Comment by JC5** [comment]
Budgets do not need an amount to be set, so you could use that. Aggregate reports for categories are on my list of things to build :)

**Reply by R-Nabil** [reply]
Yes indeed but then it seems I cant see the monthly budget spent ? If I go into the budget it shows all spendings

**Reply by R-Nabil** [reply]
Is there an ETA for this report btw ?

**Reply by pvieira84** [reply]
In my case, I use accounts for that - for me they act like parent categories and give me an outlook of the different group of expenses

**Reply by R-Nabil** [reply]
> In my case, I use accounts for that - for me they act like parent categories and give me an outlook of the different group of oexpenses

Does it allow you to see a group of expenses on a monthly basis ? I'll give it a try

**Reply by R-Nabil** [reply]
Actually it could have been a good workaround but unfortunately, it doesnt handle payback correctly.
Say you spend x amount in Clothes (online shopping), but return half of it. Your budget for Clothes spend should be only x/2, but now you would see an expense of x and a revenu of x/2. So not really an aggregate view of spending.

Of course, for a simple example like this one, its good enough. But when you start aggregating many transactions it hard to keep track. You can imagine also for instance, property investments where you have a mortgage faced by a rent, and you want to know how much is out of your pocket on a monthly or yearly basis. I thought this would be for budget + categories (budget = Property, category = Rent or Debt repayment).

I'm actually struggling to really fully rely on FireflyIII as I'm not sure how to use it to reach this kind of objective.

**Reply by pvieira84** [reply]
In my personal use case, I do not want to track the payback. Extending your example, You spend 100 € in clothes and then return 50 €. What I do is to split the original expense - 50 € in clothes and the returned 50€ to (cash account) and without a category. Same goes for the refund of 50€ - to (cash account) and without a category.
In the end, I do not report on cash account which will have a deposit and a withdrawal of 50€ without a category - only the withdrawal of 50€ categorized with clothes will count for my expenses.

For you other example, I use Expense accounts. For instance, I have a housing account where all the expenses with the different related categories go like mortgage, groceries, utilities, etc.

**Reply by R-Nabil** [reply]
> In my personal use case, I do not want to track the payback. Extending your example, You spend 100 € in clothes and then return 50 €. What I do is to split the original expense - 50 € in clothes and the returned 50€ to (cash account) and without a category. Same goes for the refund of 50€ - to (cash account) and without a category.
> 
> In the end, I do not report on cash account which will have a deposit and a withdrawal of 50€ without a category - only the withdrawal of 50€ categorized with clothes will count for my expenses.
> 
> 
> 
> For you other example, I use Expense accounts. For instance, I have a housing account where all the expenses with the different related categories go like mortgage, groceries, utilities, etc.

Not really. If you get (receive) a rent for that mortgage it would go in a revenue account. So you don't get a full view of that property cost/revenue. 

I'm testing something else at the moment and will revert back if it works.

**Reply by pvieira84** [reply]
Your only option are tags then - budgets only allow expenses. Or else, you can assign the same category to the mortgage and the rent and you report on the category to get your cost/revenue perspective

**Reply by R-Nabil** [reply]
I think I should give a fuller more detailed explanations.
Please consider the words categories/budgets etc below as generic terms, not FireflyIII definition of them, but this will help to explain better.

I will call :
Budget : A grouping of categories. For instance, with the property example below, the Property budget would include Rent income, as well as Mortgage Repayment. 
Category : A more granular (but still not that granular) budget. So Rent Income or Mortgage Repayment would be a category each.

What I would need after that is :
- Have a budget view per month : Every month, how much was spend in each aggregated categories, and how much was "earned".
- For each budget/per month, the ability to drilldown to the category level. 

Say, Property normally is flat (as much expense as revenue), but on a given month it's not. Drilling down by categories might help realise that a rent was missed. It's a silly example and not the one I actually need, but just a good illustration.

Similarly, you have a "Home" budget that includes the categories Rent (you pay rent), Groceries, Utilities etc. You expect this budget to be around x, but on a given month it's 20% above. You drilldown, you can see that, yes indeed, Utilities were more expensive that month because it was very hot and you used the AirCon way more than usual.

So my first approach was :
- My notion of category = FireflyIII Category
- My notion of Budget = FireflyIII Budget

But that doesn't work as indeed, you don't see a monthly budget view, and income don't have a budget.

Second approach I just tested seems better but still lacks a bit :
- My notion of category = Expense Account
- My notion of budget = Category

But in the Category View, you can't see the total by Expense Account (unless I missed it ?). So my "Property" category shows the sum of mortgage repayment and rent received, but I dont see them into separated summed amount. I only have every detailed transactions.


Hope this explanation is a bit better. It looks to me like this is a reasonable way of doing budgeting/accounting, so I'm pretty sure FireflyIII supports this one way or another, but I'm just not sure how :)

**Reply by 42demian42** [reply]
Hello - I was just having the "original idea" to use expense accounts as a sub-category.  So the opposite of your second approach.  

My notion of category = Category
My notion of sub category  = Expense Account

How have you been doing with your second approach ???
